Switching to use myusa and adding a manifest.

DavidEBest opened the pull request.

afeld commented on app.rb:
```diff
@@ -25,16 +26,31 @@ def initialize
+
+  get '/auth/myusa/callback' do
+    if env['omniauth.auth']
+      session[:authenticated] = true
```
The session is secure, right? i.e. someone couldn't just set this themselves? Also, I'd maybe store something like `user_email` instead, which I could see being useful in the future.

DavidEBest replied:
It is supposed to be secure, but I'll switch it out for `encrypted_cookie`, which is supposed to be more secure. Good call on the user_email bit, I'll add that.

afeld commented:
Mind adding instructions about how to run the app, including things like the required environment variables for MyUSA, and what org/space it's deployed to?

afeld commented on Procfile:
```diff
@@ -0,0 +1 @@
+web: rackup -p $PORT -E production
```
I'd get rid of this file and move this to the `command` in the `manifest.yml`.

DavidEBest commented:
@afeld I addressed your comments. There isn't a publicly accessible version of the app yet, I need to go through the process of getting the MyUSA app 'public'.

afeld commented on app.rb:
```diff
@@ -7,9 +8,9 @@ class ComplianceData < Sinatra::Base
-      username == settings.site_user && password == settings.site_password
+  helpers do
+    def authed?
+      session[:authenticated]
```
Maybe just use `!!session[:user_email]`? Then there's no need for `session[:authenticated]`.

afeld commented on README.md:
````diff
@@ -2,6 +2,41 @@
+ruby -rsecurerandom -e "puts SecureRandom.hex(32)"
+```
+
+Run the application with `rackup`
````
Missing a period here.

afeld commented:
Just two small things!

DavidEBest commented:
@afeld Thanks for the feedback. I've addressed both of those things.

afeld commented:
Needs manual merge, but otherwise, :shipit: !
